engine/src/exec-files.cpp

```diff
@@ -2510,7 +2532,18 @@ void MCFilesExecWriteToProcess(MCExecContext& ctxt, MCNameRef p_process, MCStrin
     if (MCprocesses[t_index].encoding != EN_BINARY)
 	{
 		MCStringRef t_text_data;
-		/* UNCHECKED */ MCStringConvertLineEndingsFromLiveCode(p_data, t_text_data);
+        if (!MCStringNormalizeLineEndings(p_data, 
+#ifdef __CRLF__
```

Comment:
The original ConvertLineEndinsFromLiveCode also checked for __CR__ and used CR line-endings in that case (this is the default Mac behavior).